### PR TITLE
MGMT-4893: Add default values and doc to MUST_GATHER_IMAGES in kube-api

### PR DESCRIPTION
**Pull request by slaviered**
- Add default values to agentserviceconfig in the samples directory
   - Add documentation to the new Spec field (mustGatherImages)
   - Add documentation to the code

   see the main implementation in  https://github.com/openshift/assisted-service/pull/2338

# Assisted Pull Request

## Description

  - Add default values to agentserviceconfig in the samples directory
   - Add documentation to the new Spec field (mustGatherImages)
   - Add documentation to the code

   see the main implementation in  https://github.com/openshift/assisted-service/pull/2338

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/assign @djzager 
/assign @flaper87 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/agentserviceconfig_controller.go [hunk 1]
```diff
@@ -1054,6 +1067,20 @@ func (r *AgentServiceConfigReconciler) getMustGatherImages(log logrus.FieldLogge
...
+//    also the fallback behavior in case there are no valid versions
+//    in the Spec
+//
+// 2. if osImages field is present in the AgentServiceConfig's Spec it
+//    converts the structure to the one that can be recognize by the service
+//    and returns it as a JSON string
+//
+// 3. In case both sources are present, the Spec values overrides the env
+//    values
 func (r *AgentServiceConfigReconciler) getOpenshiftVersions(log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) string {
```
Comment (djzager) [inline internal/controller/controllers/agentserviceconfig_controller.go:1084]: :grin: thank you.

**Review by djzager — APPROVED** [review]

**Comment by djzager** [comment]
@slaviered looks like the only thing you need is to `skipper make generate-all` and commit the change to `config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml` 

I wouldn't have guess that the bundle generation would reorder the sample config. Ping me when that's updated and if I'm around on Sunday I'll give it a :+1:

**Comment by YuviGold** [comment]
I'd like to sure it doesn't break the disconnected workflow
/test e2e-metal-assisted-operator-disconnected

**Review by YuviGold — APPROVED** [review]

**Pull request merged**